addons/html_builder/static/src/builder/snippets_menu_tabs/invisible_elements_tab.js

```diff
@@ -0,0 +1,104 @@
+import { Component, onWillStart, onWillUpdateProps, useState } from "@odoo/owl";
+import { WithSubEnv } from "../builder_helpers";
+import { getSnippetName } from "../../utils";
+export class InvisibleElementsTab extends Component {
+    static template = "html_builder.InvisibleElementsTab";
+    static components = { WithSubEnv };
+    static props = {
+        invisibleEls: { type: Array },
+        invisibleSelector: { type: String },
+    };
+
+    setup() {
+        this.state = useState({ invisibleEls: null });
+
+        onWillStart(() => this.updateInvisibleElementsPanel(this.props.invisibleEls));
+
+        onWillUpdateProps(async (nextProps) => {
+            this.updateInvisibleElementsPanel(nextProps["invisibleEls"]);
+        });
+    }
+
+    updateInvisibleElementsPanel(invisibleEls) {
+        // descendantPerSnippet: a map with its keys set to invisible
+        // snippets that have invisible descendants. The value corresponding
+        // to an invisible snippet element is a list filled with all its
+        // descendant invisible snippets except those that have a closer
+        // invisible snippet ancestor.
+        const descendantPerSnippet = new Map();
+        // Filter the invisibleEls to only keep the root snippets
+        // and create the map ("descendantPerSnippet") of the snippets and
+        // their descendant snippets.
+        const rootInvisibleSnippetEls = invisibleEls.filter((invisibleSnippetEl) => {
+            const ancestorInvisibleEl = invisibleSnippetEl.parentElement.closest(
+                this.props.invisibleSelector
+            );
+            if (!ancestorInvisibleEl) {
+                return true;
+            }
+            const descendantSnippets = descendantPerSnippet.get(ancestorInvisibleEl) || [];
+            descendantPerSnippet.set(ancestorInvisibleEl, [
+                ...descendantSnippets,
+                invisibleSnippetEl,
+            ]);
+            return false;
+        });
+        // Insert an invisible snippet in its "parentEl" element.
+        const createInvisibleElement = (invisibleSnippetEl, isRootParent, isDescendant) => {
```

Comment:
I would inline the object directly inside the `createInvisibleElements` as this function is only used there and is very simple.
But as it is aesthetic choice, I let you chose.